### PR TITLE
Add spring boot formatter and checkstyle

### PR DESCRIPTION
**Pull request by maiksensi**
**What**:

Introduce google style guide as discused in #133 

**Why**:

see #133 - short: error prone formatting, avoid nitpicking

**How**:

- use checkstyle maven plugin and checkstyle dependency + configuration to enable google_checks

**Checklist**:

- [x] Documentation added
      
- [x] Changelog updated
      
- [x] Tests added "N/A"
      
- [x] Ready to be merged

### CODE REVIEW

File: chaos-monkey-spring-boot/src/main/java/de/codecentric/spring/boot/chaos/monkey/assaults/ExceptionAssault.java [hunk 1]
```diff
@@ -26,7 +26,7 @@
 /** @author Thorsten Deelmann */
 public class ExceptionAssault implements ChaosMonkeyRequestAssault {
 
-  private static final Logger LOGGER = LoggerFactory.getLogger(ExceptionAssault.class);
+  private static final Logger Logger = LoggerFactory.getLogger(ExceptionAssault.class);
```
Comment (fletchgqc) [inline chaos-monkey-spring-boot/src/main/java/de/codecentric/spring/boot/chaos/monkey/assaults/ExceptionAssault.java]: Actually we should use Lombok to get the logger, but we can do that on another PR.

File: pom.xml [hunk 2]
```diff
@@ -15,61 +15,69 @@
...
-  <description>Chaos Monkey for Spring Boot parent project</description>
-
+<project xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xmlns="http://maven.apache.org/POM/4.0.0"
+  xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
 
   <properties>
     <!-- project version -->
-    <revision>2.3.0-SNAPSHOT</revision>
+    <asciidoctor-maven-plugin.version>1.6.0</asciidoctor-maven-plugin.version>
```
Comment (WtfJoke) [inline pom.xml]: Was the revision deleted on purpose? And if so the comment would be wrong.

Else changes looks good and could be merged. 👍  I‘ll aprove it but question frim above should ve answered before we merge it.

Comment (maiksensi) [reply pom.xml]: The revision should only have been moved because of the formatting. Can you double check its there? I think it is now in: Line 61

Comment (maiksensi) [reply pom.xml]: @WtfJoke ping

**Comment by maiksensi** [comment]
> Do we need to provide a link to a guide which shows how to install the code style?

Did you see: https://github.com/codecentric/chaos-monkey-spring-boot/pull/160/files#diff-bf6c05f66a41e4a8d614764c22eaf178R56 it includes a general explanation and also an explanation for intellij

File: pom.xml (hunk 2 above)
Comment (WtfJoke) [reply pom.xml]: Yes youre right, its there. Sorry didnt have time earlier. 😞

**Comment by WtfJoke** [comment]
> Did you see: https://github.com/codecentric/chaos-monkey-spring-boot/pull/160/files#diff-bf6c05f66a41e4a8d614764c22eaf178R56 it includes a general explanation and also an explanation for intellij

Yeah somehow I have seen it but ignored. Lets get it merged (as soon as latest changed are merged into)!

Sorry, that it took a while to review your last comments.
